components/Book.vue

```diff
@@ -1,17 +1,47 @@
 <template>
-    <nuxt-link :to="'/books/' + id">
-        <div class="single-book">
+    <div class="single-book">
+      <button class="fav-button" @click="favouritesEngine"></button>
+      <nuxt-link :to="'/books/' + id">
             <h2>{{title}}</h2>
             <p>{{author}}</p>
             <p>{{description}}</p>
             <div class="book-img" :style="{backgroundImage: 'url('+ img +')'}"></div>
-        </div>
-    </nuxt-link>
+      </nuxt-link>
+    </div>
 </template>
 
 <script>
 export default {
-  props: ['title', 'author', 'description', 'img', 'id']
+  data () {
+    return {
+      isFavourite: false
+    }
+  },
+  props: ['title', 'author', 'description', 'img', 'id', 'nrFav'],
+  methods: {
+    addFavourite () {
```

Comment:
This way you just unnecessarily duplicate data. I think the logic would be much easier / more effective if every book simply had predicate isFavourite property. Then adding / removing would only take one operation. In addition to that you could easily _.filter() books to get favourite ones.
Anyone correct me if I'm wrong, please.

Reply:
Agree with you